pkg/client/cmd/deploy.go

```diff
@@ -161,13 +166,34 @@ func deployApp(cmd *cobra.Command, args []string) {
 		}
 	}
 
-	ip, err := getIgnorePatterns(appFolder)
+	path := appURL
+	if url.Scheme(appURL) != "" {
+		path, err = url.FetchToTemp(appURL)
+		if err != nil {
+			client.PrintErrorAndExit("Error downloading url: %v", err)
+		}
+		defer os.Remove(path)
+	}
+
+	finfo, err := os.Stat(path)
+	if err != nil {
+		client.PrintErrorAndExit("Error accessing path: %v", err)
+	}
+	if finfo.Mode().IsRegular() {
+		path, err = tar.ExtractToTemp(path)
+		if err != nil {
+			client.PrintErrorAndExit("Error extracting archive: %v", err)
+		}
+		defer os.RemoveAll(path)
+	}
```

Comment:
What do you think about to break this code snippet in a small function?

Reply:
Yeap, deploy is already too big